src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1809,6 +1812,17 @@ public DatabaseRecord ReadDatabaseRecord()
             }
         }
 
+        private void OnCertificateChange(object sender, EventArgs e)
+        {
+            if (_requestExecutor.IsValueCreated == false) 
+                return;
+
+            using (_requestExecutor.Value)
+                _requestExecutor = CreateRequestExecutor();
+        }
+
+        private Lazy<RequestExecutor> CreateRequestExecutor() => new(() => RequestExecutor.CreateForServer(new[] { ServerStore.Server.WebUrl }, Name, ServerStore.Server.Certificate.Certificate, DocumentConventions.DefaultForServer), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
Would it work if `WebUrl` is set to `0.0.0.0`?

I know it is an old code, but something I've encountered and we need to check it here as well

Reply:
```
        private string GetWebUrl(string kestrelUrl)
        {
            var serverUri = new Uri(Configuration.Core.ServerUrls[0]);
            if (serverUri.IsDefaultPort == false && serverUri.Port == 0)
            {
                var kestrelUri = new Uri(kestrelUrl);
                return new UriBuilder(serverUri)
                {
                    Port = kestrelUri.Port
                }.Uri.ToString();
            }

            return Configuration.Core.ServerUrls[0];
        }
```

I do not think this is possible to get `0.0.0.0` here.

Reply:
using GetNodeHttpServerUrl